snuba/web/db_query.py

```diff
@@ -875,18 +875,19 @@ def _apply_allocation_policies_quota(
                 allowance = allocation_policy.get_quota_allowance(
                     attribution_info.tenant_ids, query_id
                 )
-                num_threads = min(num_threads, allowance.max_threads)
                 can_run &= allowance.can_run
                 quota_allowances[allocation_policy.config_key()] = allowance
                 span.set_data(
                     "quota_allowance",
                     quota_allowances[allocation_policy.config_key()],
                 )
                 if allowance.is_throttled:
-                    throttle_quota_and_policy = _QuotaAndPolicy(
-                        quota_allowance=allowance,
-                        policy_name=allocation_policy.config_key(),
-                    )
+                    if allowance.max_threads < num_threads:
```

Comment:
These two conditions can be `and`ed. For readability, I would rename `num_threads` to `min_threads_across_policies` or a shorter version of it. `num_threads` is a bit abstract and does not really call out the intent of this change.